src/extension/background-script/actions/onboard/providerOnboard.ts

```
@@ -0,0 +1,32 @@
+import utils from "~/common/lib/utils";
+import { getHostFromSender } from "~/common/utils/helpers";
+import type { MessageAllowanceEnable, Sender } from "~/types";
+
+const providerOnboard = async (
+  message: MessageAllowanceEnable,
+  sender: Sender
+) => {
+  const host = getHostFromSender(sender);
+  if (!host) return;
+
+  try {
+    const response = await utils.openPrompt<{
+      enabled: boolean;
+      remember: boolean;
+    }>(message);
+
+    return {
```

I think we should return an error here, as we will not get this data